### DISCUSSION TITLE
Pre-built .apk with offline bootstrap?

### DISCUSSION THREAD

**Discussion by Unknown78** [nix-community/nix-on-droid#242] [Ideas]
How do I get pre-built .apk with bootstrap included? I need it in the case of no internet connection.

I run on marshmallow 6.0.1 and latest termux is only for nougat v7+.
The legacy repository has bootstrap included as `termux-v0.79-offline-bootstraps.apk`
https://archive.org/download/termux-repositories-legacy

Could I have something like `com.termux.nix_188031-offline-bootstrap-release-22.11.apk`?

**Comment by t184256** [comment]
You might get away with specifying a local URL / path to the bootstrap dialog, but the problem is, our bootstrap itself needs network. All in all, Nix package manager isn't exactly offline-friendly either.

**Reply by Unknown78** [reply]
will there be any plan to make it offline-friendly?

**Reply by t184256** [reply]
That's unlikely.

**Reply by Unknown78** [reply]
Okay but I'd still like to try it. How do you enter the local URL?
I've tried copying `bootstrap-aarch64.zip` to the root of  my internal storage and then enter `file://sdcard/` as the local URL. But it just throws me an error.`

**Reply by t184256** [reply]
IDK, all I know is that [this is the code](https://github.com/t184256/nix-on-droid-app/blob/master/app/src/main/java/com/termux/app/TermuxInstaller.java).

**Reply by Unknown78** [reply]
But I don't understand the code...
